src/services/dashboards/dashboard-customize/modules/DashboardAddDefaultWidgetRightArea.vue

```diff
@@ -108,20 +198,118 @@ export default defineComponent<Props>({
+            return selectedItem && !isEmpty(selectedItem);
+        };
+        const getRefinedJsonSchema = (originalJsonSchema: JsonSchema, selectedOptions, inheritItemMap) => {
+            const _jsonSchema = cloneDeep(originalJsonSchema);
+            const refinedJsonSchema: Partial<JsonSchema> = {
+                type: 'object',
+                properties: {},
+                required: state.requiredProperties,
+            };
+            Object.entries(_jsonSchema.properties).forEach(([propertyName, propertySchema]) => {
+                if (!state.requiredProperties.includes(propertyName) && !selectedOptions.find((d) => d.name === propertyName)) {
+                    return;
+                }
+                if (inheritItemMap[propertyName]) {
+                    refinedJsonSchema.properties[propertyName] = {
+                        title: (propertySchema as JsonSchema).title,
+                        ...SAMPLE_DASHBOARD_VARIABLES_SCHEMA, // TODO: temp data
+                    };
+                } else {
+                    const refinedPropertyName = propertyName.replace('filters.', '');
+                    const storeData: ReferenceItem = storeState[refinedPropertyName];
+                    if (refinedPropertyName === 'group_by') {
+                        refinedJsonSchema.properties[refinedPropertyName] = propertySchema;
+                        return;
+                    }
+                    const menuItems = Object.values(storeData).map((d) => ({
+                        name: d.key, label: d.label,
+                    }));
+                    if ((propertySchema as JsonSchema).type === 'array') {
+                        refinedJsonSchema.properties[propertyName] = {
+                            ...propertySchema as object,
+                            items: { enum: Object.keys(storeData) },
+                            menuItems,
+                            default: [],
+                        };
+                    } else {
+                        refinedJsonSchema.properties[propertyName] = {
+                            ...propertySchema as object,
+                            enum: Object.keys(storeData),
+                            menuItems,
+                            default: undefined,
+                        };
+                    }
+                }
+            });
+            return refinedJsonSchema;
+        };
+
         /* Event */
+        const hideOptionsMenu = () => {
+            state.addOptionsMenuVisible = false;
+        };
         const handleChangeInheritToggle = (propertyName: string, { value }) => {
-            state.inheritFormData[propertyName] = value;
+            // init form
+            state.formData[propertyName] = undefined;
+            // update inherit data
+            const _inheritItemMap = cloneDeep(state.inheritItemMap);
+            _inheritItemMap[propertyName] = value;
+            state.inheritItemMap = _inheritItemMap;
```

Comment:
question: Why is it required?

Reply:
Because `state.inheritItemMap` is used in `computed`.
But, I think this should be changed...